### PR TITLE
Adding calculation functionality for number and decimal number

### PR DESCRIPTION
**Pull request by sahnavin123**
# Description: 
This is a simple calculator that allows the user to perform basic arithmetic operations such as addition, subtraction, multiplication, and division. The calculator has a clear button that clears the display and resets all the variables. The backspace button allows the user to delete the last digit entered.

## Features:
- User can enter numbers using the number buttons (0-9)
- User can enter decimal points using the "." button
- User can perform basic arithmetic operations (+, -, *, /) using the operator buttons
- User can clear the display and reset all the variables using the "Clear" button
- User can delete the last digit entered using the backspace button
- User can get the result of the arithmetic operation by clicking the "=" button

## Manual Testing:
- Entered a number by clicking on one of the number buttons (0-9).
- Entered a decimal point by clicking on the "." button.
- Entered an operator (+, -, *, /) by clicking on one of the operator buttons.
- Entered a second number by clicking on one of the number buttons (0-9).
- Clicked on the "=" button to get the result of the arithmetic operation.
- Clicked on the "Clear" button to clear the display and reset all the variables.
- Clicked on the backspace button to delete the last digit entered.
- Performed different arithmetic operations using the calculator and verified that the results were correct.

## Page screenshot:

| image  | image  | image  |
|:-:|:-:|:-:|

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        firstNumber = result.toString();
+        operator = "";
+        secondNumber = "";
+      }
+    } else {
+      operator = buttonValue;
+      updateDisplay(buttonValue);
+    }
+  });
+});
```
Comment (devanuj97) [inline script.js]: logic is convoluted, please refactor as discussed 
you can user  `parseFloat` to obtain float value from string wether that is present as float or int
Example -
parseFloat("1") => 1
parseFloat("1.2") => 1.2

File: script.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  if (display.textContent === "0" || result !== "") {
+    display.textContent = value;
+    result = "";
+  } else {
+    display.textContent += value;
+  }
+};
+
+const backspace = () => {
+  if (secondNumber !== "") {
```
Comment (devanuj97) [inline script.js]: can we avoid if-else ladder here or other places let's give a thought on this, what happen when if-else ladder is it decrease the readability of code this becomes hard to go into each nested node if condition and meshed up with logic, this should be only our last option to opt

File: script.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+      return a - b;
+    case "*":
+      return a * b;
+    case "/":
+      if (b === 0) {
+        return "Cannot divide by zero!";
+      } else {
+        return a / b;
+      }
+  }
```
Comment (devanuj97) [inline script.js:23]: can we use ternary operator here ?

**Comment by devanuj97** [comment]
Logic is wrong @sahnavin123 did you tested your self ? i don't know why you are being so careless about this at least test all the logic and corner case by your self first this is the 2nd or 3rd time i'm pointing out the logical issue on this specific PR
Try evaluating this expression -- 1+2-1

**Review by devanuj97 — APPROVED** [review]
